### PR TITLE
Add dataset histograms and truncation suggestions

### PR DESCRIPTION
**Pull request by rasbt**
As discussed, this adds max-seq-length recommendations @carmocca

### CODE REVIEW

File: tutorials/prepare_dataset.md [hunk 1]
```diff
@@ -45,9 +44,25 @@ python scripts/prepare_alpaca.py \
...
+
+In this case, a cut-off of 256 may be a reasonable choice:
+
+```bash
+python scripts/prepare_alpaca.py \
+ --checkpoint_dir checkpoints/tiiuae/falcon-7b \
+ --max_seq_length 256
+```
+
+For comparison, the Falcon 7B model requires xx GB of memory for the original Alpaca dataset and xx GB of memory for the truncated Alpaca dataset when finetuning with LoRA using a micro batchsize of 1 and bfloat-16 precision.
```
Comment (carmocca) [inline tutorials/prepare_dataset.md:61]: LGTM! We can merge this once this is filled in

File: tutorials/prepare_dataset.md [hunk 2]
```diff
@@ -45,9 +44,25 @@ python scripts/prepare_alpaca.py \
  --checkpoint_dir checkpoints/tiiuae/falcon-7b
 ```
 
+#### Truncating datasets
```
Comment (carmocca) [inline tutorials/prepare_dataset.md:47]: Can you link to this section from the oom.md page?

Comment (rasbt) [reply tutorials/prepare_dataset.md:47]: Good idea. Fixed that!

File: tutorials/prepare_dataset.md (hunk 1 above)
Comment (rasbt) [reply tutorials/prepare_dataset.md]: Done!

**Review by carmocca — APPROVED** [review]

**Pull request merged**